src/Farmer/Common.fs

```diff
@@ -1833,15 +1833,15 @@ module Identity =
 
     /// Represents a User Assigned Identity, and the ability to create a Principal Id from it.
     type UserAssignedIdentity =
-        | UserAssignedIdentity of LinkedResource
+        | UserAssignedIdentity of ResourceId
+        | LinkedUserAssignedIdentity of LinkedResource
```

Comment:
This can be LinkedUserAssignedIdentity of ResourceId.
The only use case for using the LinkedUserAssignedIdentity of LinkedResource I can think of is if we need to deprecate UserAssignedIdentity in the future.